backend/src/main/java/com/woowacourse/kkogkkog/application/AuthService.java

```diff
@@ -1,32 +1,34 @@
 package com.woowacourse.kkogkkog.application;
 
 import com.woowacourse.kkogkkog.application.dto.TokenResponse;
-import com.woowacourse.kkogkkog.domain.Member;
-import com.woowacourse.kkogkkog.domain.repository.MemberRepository;
-import com.woowacourse.kkogkkog.exception.member.MemberNotFoundException;
-import com.woowacourse.kkogkkog.exception.member.MemberWrongInputException;
-import com.woowacourse.kkogkkog.presentation.dto.TokenRequest;
+import com.woowacourse.kkogkkog.application.dto.MemberCreateResponse;
+import com.woowacourse.kkogkkog.infrastructure.SlackRequester;
+import com.woowacourse.kkogkkog.infrastructure.SlackUserInfo;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 @Transactional
 @Service
 public class AuthService {
 
-    private final MemberRepository memberRepository;
     private final JwtTokenProvider jwtTokenProvider;
+    private final MemberService memberService;
+    private final SlackRequester slackRequester;
 
-    public AuthService(MemberRepository memberRepository, JwtTokenProvider jwtTokenProvider) {
-        this.memberRepository = memberRepository;
+    public AuthService(JwtTokenProvider jwtTokenProvider, MemberService memberService,
+        SlackRequester slackRequester) {
         this.jwtTokenProvider = jwtTokenProvider;
+        this.memberService = memberService;
+        this.slackRequester = slackRequester;
     }
 
-    public TokenResponse login(TokenRequest tokenRequest) {
-        Member findMember = memberRepository.findByEmail(tokenRequest.getEmail())
-            .orElseThrow(MemberNotFoundException::new);
-        if (findMember.isNotSamePassword(tokenRequest.getPassword())) {
-            throw new MemberWrongInputException();
-        }
-        return new TokenResponse(jwtTokenProvider.createToken(findMember.getId().toString()));
+    @Transactional(readOnly = true)
+    public TokenResponse login(String code) {
+        SlackUserInfo userInfo = slackRequester.getUserInfoByCode(code);
+        MemberCreateResponse memberCreateResponse = memberService.saveOrFind(userInfo);
```

Comment:
member가 저장될 수도 있으니 readOnly는 뺍시다.

```suggestion
    public TokenResponse login(String code) {
        SlackUserInfo userInfo = slackRequester.getUserInfoByCode(code);
        MemberCreateResponse memberCreateResponse = memberService.saveOrFind(userInfo);
```